src/sim/syscall_emul.hh

```diff
@@ -677,6 +678,68 @@ copyOutStatfsBuf(TgtStatPtr tgt, HostStatPtr host)
 #endif
 }
 
+template <typename OS, typename TgtStatPtr, typename HostStatPtr>
+void
+copyOutStatxBuf(TgtStatPtr tgt, HostStatPtr host, bool fakeTTY = false)
+{
+#if !defined(__linux__)
+    static_assert(false, "statx is a Linux-specific system call.");
+#endif
```

Comment:
Can I ask you why you are checking if we are in a linux host?

This is checked in other stat* implementations (like ststfs) when the *host* syscall is used to mimic the guest (linux) syscall. In this case it doesn't seem like the backend of statxFunc is using statx so I don't see the reason why we should check this, especially since the static_assert will fail at compile time making it impossible to build on non-Linux hosts (like macOS)

Reply:
I understand, thanks for explaining. I removed this check in the latest commit.